specs/EnableInputPassthrough.md

```diff
@@ -0,0 +1,102 @@
+# Background
+WebView2 .NET developers often encounter issues where certain default events are not raised in the WebView2 .NET control. This issue occurs because user inputs are directly delivered to the browser, and the host application does not receive the corresponding messages. This causes some default event handling API of .NET control not to work, such as PreProcessMessage and ProcessCmdKey in WinForms. It also prevents some keys from reaching Key Event realated APIs such as OnKeyDown and ToolStripMenuItem. In order to solve this type of issue, we provide an api which allows user inputs pass through the browser to the host app.
+
+# Description
+`EnableInputPassthrough` allows user input to pass through the browser and allows inputs received in the host app process. The messages can be received by Win32 API ::GetMessage() or ::PeekMessage(). This provides the host app a chance to handle the message before dispatching to the WebView2 HWND.
+Different from visual hosting, user inputs are dispatched to the HWND owned by CoreWebView2Controller using EnableInputPassthrough, it is not required for developers to send user inputs from the app to the WebView2 control. It is not expected to use EnableInputPassthrough together with visual hosting.
+
+# Examples
+## Win32 C++
+```cpp
+HRESULT AppWindow::CreateControllerWithInputPassthrough()
+{
+      //! [CreateControllerWithOptions]
+    auto webViewEnvironment10 = m_webViewEnvironment.try_query<ICoreWebView2Environment10>();
+    if (!webViewEnvironment10)
+    {
+        FeatureNotAvailable();
+        return S_OK;
+    }
+
+    wil::com_ptr<ICoreWebView2ControllerOptions> options;
+    // The validation of parameters occurs when setting the properties.
+    HRESULT hr = webViewEnvironment4->CreateCoreWebView2ControllerOptions(options.GetAddressOf());
+    if (hr == E_INVALIDARG)
+    {
+        ShowFailure(hr, L"Unable to create WebView2 due to an invalid profile name.");
```

Comment:
Can CreateCoreWebView2ControllerOptions fail with E_INVALIDARG? It seems like you have passed no 'in' parameters to the method.

Reply:
No, I copied it from sample app. Removed this check.